test/test_tracer.cpp

```diff
@@ -966,32 +1014,22 @@ TEST_CASE("create 128-bit trace IDs") {
   config.extraction_styles.push_back(PropagationStyle::B3);
   const auto finalized = finalize_config(config);
   REQUIRE(finalized);
-  Tracer tracer{*finalized};
+  Tracer tracer{*finalized, clock};
+  TraceID trace_id;  // used below the following SECTIONs
 
   SECTION("are generated") {
+    // Specifically, verify that the high 64 bits of the generated trace ID
+    // contain the unix start time of the trace shifted up 32 bits.
+    //
+    // Due to the definition of `clock`, above, that unix time will be
+    // `flash_crash`.
     const auto span = tracer.create_span();
-    // The chance that it's zero is ~2**(-64), which I'm willing to neglect.
-    REQUIRE(span.trace_id().high != 0);
-  }
-
-  SECTION("result in _dd.p.tid trace tag being sent to collector") {
-    TraceID generated_id;
-    {
-      const auto span = tracer.create_span();
-      generated_id = span.trace_id();
-    }
-    CAPTURE(logger->entries);
-    REQUIRE(logger->error_count() == 0);
-    REQUIRE(collector->span_count() == 1);
-    const auto& span = collector->first_span();
-    const auto found = span.tags.find(tags::internal::trace_id_high);
-    REQUIRE(found != span.tags.end());
-    const auto high = parse_uint64(found->second, 16);
-    REQUIRE(high);
-    REQUIRE(*high == generated_id.high);
+    const std::uint64_t expected = std::uint64_t(flash_crash) << 32;
```

Comment:
I'd probably compare this with a raw expected value instead of a computed one.

Reply:
That's a fine line and I see your point. The test is kind of circular if it just does what the implementation does.

But if I were to include the raw expected value in the test, I'd obtain it by calculating `1273171513 << 32` (which is `5468230010533838848`). At that point we'd be testing that `operator<<(uint64_t, int)` behaves the same way for all build targets, which is meh.